frontend/src/logic/schedule-logic/public-holidays.logic.ts

```diff
@@ -0,0 +1,70 @@
+// dates as represented here follow [day, month] pattern
+// months as used here are human-readable i.e. go from 1 to 12 (hence + 1)
+
+export class PublicHolidaysLogic {
+  private publicHolidayDates: number[][];
+
+  constructor(year: string) {
+    this.publicHolidayDates = [
+      // fixed holidays (święta stałe)
+      [1, 1],
+      [6, 1],
+      [1, 5],
+      [3, 5],
+      [15, 8],
+      [1, 11],
+      [11, 11],
+      [25, 12],
+      [26, 12],
+      // moveable holidays (święta ruchome)
+      ...this.moveableHolidays(parseInt(year)),
+    ];
+  }
+
+  public isPublicHoliday(day: number, month: number): boolean {
+    return (
+      this.publicHolidayDates.find((holiday) => holiday[0] === day && holiday[1] === month + 1) !==
+      undefined
+    );
+  }
+
+  private moveableHolidays(year: number): number[][] {
+    function dateToDayMonth(date: Date): number[] {
+      return [date.getDate(), date.getMonth() + 1];
+    }
+
+    // Easter
+    const holidayDate = this.calculateEasterDate(year);
+    let result = [dateToDayMonth(holidayDate)];
+
+    // Easter Monday, 1 day after Easter (Poniedziałek Wielkanocny)
+    holidayDate.setDate(holidayDate.getDate() + 1);
+    result = [...result, dateToDayMonth(holidayDate)];
+
+    // Green Week, 49 days after Easter (Zielone Świątki)
+    holidayDate.setDate(holidayDate.getDate() + 48);
+    result = [...result, dateToDayMonth(holidayDate)];
+
+    // Feast of Corpus Christi, 60 days after Easter (Boże Ciało)
+    holidayDate.setDate(holidayDate.getDate() + 11);
+    result = [...result, dateToDayMonth(holidayDate)];
+
+    return result;
+  }
+
+  // refer to https://pl.wikipedia.org/wiki/Wielkanoc#Metoda_Meeusa/Jonesa/Butchera
+  private calculateEasterDate(year: number): Date {
```

Comment:
Piękne to jest XD

Reply:
Przydały by się testy xD
Szczególnie teraz, jak był zmergowany ten MR
https://github.com/Project-Summer-AI-Lab-Glider/nurse-scheduling-problem-frontend/pull/46

Reply:
W sumie to Idealny use case XD To od teraz wszyscy piszemy teściki?

Reply:
Tu napewno się przydadzą. I w kolejnych PR, myślę że też. 
Głównie to zacząć.

Reply:
Tylko nie wiem, czy nie zrobić osobne zadanie na przetestowanie całości.
Czy może po prostu przyjmiemy, że zaczynając od tego PR ( pomijając ten https://github.com/Project-Summer-AI-Lab-Glider/nurse-scheduling-problem-frontend/pull/57 XD ) zawsze piszemy testy.

Reply:
bedo testy